core/src/test/java/greencity/controller/HabitStatisticControllerTest.java

```diff
@@ -0,0 +1,120 @@
+    @InjectMocks
+    private HabitStatisticController habitStatisticController;
+    private UserVO userVO = new UserVO();
+    private ObjectMapper objectMapper = new ObjectMapper();
+    private final Long id = 1l;
+
+    @BeforeEach
+    void setUp(){
+        objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+        this.mockMvc = MockMvcBuilders.standaloneSetup(habitStatisticController)
+                .build();
+    }
+
+    @Test
+    void findAllByHabitId_HabitStatisticControllerTest_shouldReturnIsOkStatus() throws Exception{
+        mockMvc.perform(get(habitStatisticControllerLink + "/{habitId}", id)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        verify(habitStatisticService).findAllStatsByHabitId(id);
+    }
+
+    @Test
+    void findAllStatsByHabitAssignId_HabitStatisticControllerTest_shouldReturnIsOkStatus() throws Exception{
+        mockMvc.perform(get(habitStatisticControllerLink + "/assign/{habitAssignId}", id)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        verify(habitStatisticService).findAllStatsByHabitAssignId(id);
+    }
+
+    @Test
+    void saveHabitStatistic_HabitStatisticControllerTest_shouldReturnIsCreatedStatus() throws Exception{
+        String addHabitStatisticDtoToString = "{\"amountOfItems\":10,\"habitRate\":\"GOOD\",\"createDate\":\"" + ZonedDateTime.now().toString() + "\"}";
+        when(habitStatisticService.saveByHabitIdAndUserId(eq(id), eq(userVO.getId()), any()))
+                .thenReturn(new HabitStatisticDto());
+        mockMvc.perform(post(habitStatisticControllerLink + "/{habitId}", id)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(addHabitStatisticDtoToString))
+                .andExpect(status().isCreated());
+        verify(habitStatisticService).saveByHabitIdAndUserId(eq(id), eq(userVO.getId()), any());
+    }
+    @Test
+    void updateStatistic_HabitStatisticControllerTest_shouldReturnIsOkStatus() throws Exception{
+        UpdateHabitStatisticDto updateDto = UpdateHabitStatisticDto.builder()
+                .amountOfItems(1)
+                .habitRate(HabitRate.GOOD)
+                .build();
+        when(habitStatisticService.update(anyLong(), anyLong(), eq(updateDto)))
+                .thenReturn(new UpdateHabitStatisticDto());
+        mockMvc.perform(put(habitStatisticControllerLink + "/{habitId}", id)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(updateDto)))
+                .andExpect(status().isOk());
+        verify(habitStatisticService).update(anyLong(), anyLong(), eq(updateDto));
+    }
+
+   @Test
+    void getTodayStatisticsForAllHabitItems_HabitStatisticControllerTest_shouldReturnIsOkStatus() throws Exception{
+        Locale locale = new Locale("en");
```

Comment:
Constructor `Locale(java.lang.String)` is deprecated since Java 19 . Please use static method `Locale.of("en")`